### PR TITLE
36 GitHub actions junit tests

### PR DESCRIPTION
**Pull request by any1-source**
added Junit tests.

I bugfixed the methods, where I implemented tests for If bugs where found.

### CODE REVIEW

File: backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
     private int bookingIdentifier;
 
     @Get("txt")
-    public String toString() {
+    public String toString() throws ResourceException {
         checkInRole(ROLE_SECURITY);
         //SELECT * FROM bookings where id = {bookingIdentifier}
         BookingsRecord result = dslContext.fetchOne(BOOKINGS, BOOKINGS.ID.eq(bookingIdentifier));
 
-        return result != null? result.formatJSON(jSONFormat) : null;
```
Comment (N0ah-S) [inline backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java:24]: Exhibit A: Missing ' '

File: backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java [hunk 2]
```diff
@@ -35,17 +32,16 @@ protected void doInit() throws ResourceException {
...
 
         String result = this.toString();
 
+
+        if (Integer.parseInt(getClientInfo().getUser().getIdentifier()) != bookingIdentifier && !isInRole("admin"))
+            throw new ResourceException(403);
         //DELETE bookings where id = {bookingIdentifier}
-        if(Integer.parseInt(getClientInfo().getUser().getName()) == bookingIdentifier  || isInRole("admin")){
-            dslContext.delete(BOOKINGS).where(BOOKINGS.ID.eq(bookingIdentifier)).execute();
-        }else{
```
Comment (N0ah-S) [inline backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java:46]: Exhibit B

File: backend/src/main/java/de/cae/XYFleet/ressource/UserRessource.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        checkInRole(ROLE_ADMIN);
+
+        //get Query entries
+        Form form = getQuery();
+        Map<String, String> values = form.getValuesMap();
+
+        //Assert input
+        String[] expected = {"name", "passwort", "is_driver", "role"};
+        //check if all expected values are given
+        for(String a : expected) if(values.get(a) == null) throw new ResourceException(400, "column: "+ a +", missing value.");
```
Comment (N0ah-S) [inline backend/src/main/java/de/cae/XYFleet/ressource/UserRessource.java:40]: Exhibit C

File: backend/src/main/java/de/cae/XYFleet/ressource/UserRessource.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        Form form = getQuery();
+
+        UpdateSetFirstStep<UsersRecord> firstStep = dslContext.update(USERS);
+        UpdateSetMoreStep<UsersRecord> moreStep = null;
+
+        for (Parameter param : form) {
+            if(param.getValue() == null) throw new ResourceException(400, "no value for given param: "+param.getName());
+            moreStep = switch (param.getName()) {
+                case "name" -> firstStep.set(USERS.NAME, param.getValue());
+                case "passwort" -> firstStep.set(USERS.PASSWORT, param.getValue());
```
Comment (N0ah-S) [inline backend/src/main/java/de/cae/XYFleet/ressource/UserRessource.java:79]: password - seriously: just do a replace operation, mixing german and english in variable names is dangerous

File: backend/src/main/java/de/cae/XYFleet/ressource/UserRessource.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                case "role" -> firstStep.set(USERS.ROLE, param.getValue());
+                case "is_driver" -> firstStep.set(USERS.IS_DRIVER, Byte.parseByte(param.getValue()));
+                default -> throw new ResourceException(400, "Invalid param in query: " + param.getName(), "Invalid param in query: " + param.getName());
```
Comment (N0ah-S) [inline backend/src/main/java/de/cae/XYFleet/ressource/UserRessource.java:82]: why do the same thing twice and that only here?
````java 
//82
new ResourceException(400, "Invalid param in query: " + param.getName(), "Invalid param in query: " + param.getName());
//87
throw new ResourceException(400, "nothing to do. no params in query given");

File: backend/src/test/java/UserRessourceTest.java [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+import static de.cae.XYFleet.authentication.XYAuthorizer.*;
+import static de.cae.XYFleet.ressource.XYServerResource.jSONFormat;
+import static org.jooq.codegen.XYFleet.Tables.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class UserRessourceTest extends RessourceTest {
+    @BeforeAll
+    public static void initAll() {
+        //Arrange
+        UsersRecord user = new UsersRecord(0, "mMustermann", "123", "user", Byte.parseByte("1"));
```
Comment (N0ah-S) [inline backend/src/test/java/UserRessourceTest.java:26]: `Byte.parseByte("1")`
uhhhhhhhhhh

File: sql/init.sql [hunk 7]
```diff
@@ -1,5 +1,17 @@
+DROP SCHEMA SWT;
+
 create schema if not exists SWT;
 
+create table if not exists SWT.users
```
Comment (N0ah-S) [inline sql/init.sql:5]: Either remove the SWT.user SCHEMA below, or adjust it with this one. No need to have 'dead' code that isn't even outcommented.

**Review by N0ah-S — CHANGES_REQUESTED** [review]
# General Things
1. try reformating the whole project (or at least your changes) with your IDEs standards for consistency' sake. See the Exhibits (there are way more, I just gave up)
2. bookingIdentifier seems to be a very odd name for an user id. Please review whether you should adjust some of it's occurrences.

## See comments for everything else ##

File: backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java (hunk 1 above)
Comment (any1-source) [reply backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java:24]: Where is a space missing??? please specifiy

File: backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java (hunk 2 above)
Comment (any1-source) [reply backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java:46]: What do you mean by eshibit B? Is it a way to tell me that it is the same Issue like Exhibit A?

File: backend/src/main/java/de/cae/XYFleet/ressource/UserRessource.java (hunk 3 above)
Comment (any1-source) [reply backend/src/main/java/de/cae/XYFleet/ressource/UserRessource.java:40]: Exhibit B

File: backend/src/main/java/de/cae/XYFleet/ressource/UserRessource.java (hunk 5 above)
Comment (any1-source) [reply backend/src/main/java/de/cae/XYFleet/ressource/UserRessource.java:82]: relict of old misconception, I couldnt find where to set the output of the errormessage.

File: backend/src/main/java/de/cae/XYFleet/ressource/UserRessource.java (hunk 4 above)
Comment (any1-source) [reply backend/src/main/java/de/cae/XYFleet/ressource/UserRessource.java:79]: done

File: backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java (hunk 1 above)
Comment (N0ah-S) [reply backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java:24]: Before the '?' - just usw Autoformat.

File: backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java (hunk 2 above)
Comment (N0ah-S) [reply backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java:46]: Exhibit B is the second example for unformated code.

Comment (N0ah-S) [reply backend/src/main/java/de/cae/XYFleet/ressource/BookingRessource.java:46]: > try reformating the whole project (or at least your changes) with your IDEs standards for consistency' sake. See the Exhibits (there are way more, I just gave up)

**Comment by N0ah-S** [comment]
Variablen schreiben wir am Anfang klein: Identifier -> identifier.

Fun Fact: 0b0 und 0b1 sind direkt Binärwerte, ist aber eigentlich egal.

**Review by N0ah-S — CHANGES_REQUESTED** [review]
Variablen schreiben wir am Anfang klein: Identifier -> identifier.

Fun Fact: 0b0 und 0b1 sind direkt Binärwerte, ist aber eigentlich egal.

**Comment by N0ah-S** [comment]
> Variablen schreiben wir am Anfang klein: Identifier -> identifier.

**Comment by N0ah-S** [comment]
```sql
DROP SCHEMA SWT;

--change to

DROP SCHEMA IF EXISTS SWT;
```

Otherwise this script will fail unless it was already executed, making new/first installs of the DB a nightmare.

**Review by N0ah-S — APPROVED** [review]
LGTM

**Pull request merged**